### PR TITLE
util/log,cli/start: assert that config changes happen before logging

### PR DESCRIPTION
**Pull request by knz**
This commit adds an assertion that configuration changes in the
logging package happen before the first event is logged.

Release note: None

### CODE REVIEW

File: pkg/util/log/clog.go [hunk 1]
```diff
@@ -122,6 +122,10 @@ type loggerT struct {
 	mu struct {
 		syncutil.Mutex
 
+		// active indicates that at least one event has been logged
+		// to this logger already.
```
Comment (irfansharif) [inline pkg/util/log/clog.go:126]: Also mention the configuration changes + secondary loggers here? (basically the comment block for IsActive, which we could just point to as well.)

File: pkg/util/log/flags.go [hunk 2]
```diff
@@ -65,6 +77,11 @@ func SetupRedactionAndStderrRedirects() (cleanup func(), err error) {
 	// won't be positioned outside of log redaction markers and
 	// mistakenly considered as "safe for reporting".
 
+	// Sanity check.
+	if IsActive() {
+		panic(errors.New("cannot set up logging after the first logging event"))
```
Comment (irfansharif) [inline pkg/util/log/flags.go:82]: Is this too specific to logging events? We could just leave it at it being "already active".

File: pkg/cli/start.go [hunk 3]
```diff
@@ -1202,6 +1202,10 @@ func logOutputDirectory() string {
 func setupAndInitializeLoggingAndProfiling(
 	ctx context.Context, cmd *cobra.Command,
 ) (stopper *stop.Stopper, err error) {
+	if log.IsActive() {
+		panic(errors.New("cannot set up logging after the first logging event"))
```
Comment (irfansharif) [inline pkg/cli/start.go:1206]: Ditto, too specific?

**Review by irfansharif — APPROVED** [review]
LGTM.

**Comment by knz** [comment]
bors r=irfansharif

**Pull request merged**